modules/ppcp-wc-gateway/src/WCGatewayModule.php

```diff
@@ -867,10 +868,16 @@ static function () use ( $container ): void {
 				$logger = $container->get( 'woocommerce.logger.woocommerce' );
 				assert( $logger instanceof LoggerInterface );
 				try {
+					$task_lists = TaskLists::get_lists();
+					if ( ! isset( $task_lists['extended'] ) ) {
+						return;
```

Comment:
Maybe it's beneficial to add this condition directly into the `TaskRegistrar` class where the 'extended' attribute is used, or would that cause other problems?

https://github.com/woocommerce/woocommerce-paypal-payments/blob/4484386e498c09bdaf24df73ad28307de747e6f5/modules/ppcp-wc-gateway/src/Settings/WcTasks/Registrar/TaskRegistrar.php#L24-L31

Reply:
Ideally, the purpose of the `TaskRegistrar` is "To Register The Task", so I think it shouldn't care about any condition. It should add or throw an error on failure. And also with early return, the rest of other code is not running.
So why it is beneficial to add it inside?

Reply:
I have two concerns about the current PR: 

1. If a different place in our plugin wants to register a "things to do next" task, that code would also need to manually check for the existence of the "extended" list before registering a task, i.e. this is a pattern that requires code duplication if we keep it
1. Since the service only asserts that the task-registrar implements a certain interface (`assert( $task_registrar instanceof TaskRegistrarInterface )`) we cannot guarantee in the service-file that the task-registrar actually uses TaskLists, or a different way to register tasks.


Short: I think checking the TaskLists::get_lists() response should be a responsibility of the Registrar class that also adds the task to that object.

Two options that, I think, would be better:
- Either add a new `$task_registrar->can_add_task()` before trying to register.
- Or, simply keep the error (nofix), since it's not a bug, but documents missing WooCommerce configuration which is outside our plugin's control

Reply:
Sorry, but I don't see a reason for other part of the code to try to register a task in a different way?

- If you want to create "Simple redirect tasks" let's say from other module, you just need to extend [this](https://github.com/woocommerce/woocommerce-paypal-payments/blob/4484386e498c09bdaf24df73ad28307de747e6f5/modules/ppcp-wc-gateway/services.php#L2036) configuration
- if you want to add "new" type of tasks you can create your own config, get that config [here](https://github.com/woocommerce/woocommerce-paypal-payments/blob/4484386e498c09bdaf24df73ad28307de747e6f5/modules/ppcp-wc-gateway/src/WCGatewayModule.php#L885) and merge with the rest of the tasks before passing to `TaskRegistrar`

The whole idea is to have a central place/way to register the tasks. 

> Or, simply keep the error (nofix), since it's not a bug, but documents missing WooCommerce configuration which is outside our plugin's control

I think if you don’t have that section, you shouldn’t add any tasks to it, so for me it’s ok to check if section exists before trying to add anything to it.

What do you think about this @Dinamiko ?
cc: @InpsydeNiklas

Reply:
Update: I am passing the `$list_id` to the `register` method and adding the check there now: 10a95b0f2e797b484d7c67896bcf1aea6b4dad42